x-pack/plugins/infra/server/routes/profiling/index.ts

```diff
@@ -48,18 +54,18 @@ export function initProfilingRoutes({ framework, getStartServices, logger }: Inf
 
   framework.registerRoute(
     {
-      method: 'post',
+      method: 'get',
```

Comment:
I've converted all endpoints to GET in order to take advantage of browser caching (with an additional header bellow in the code). This way when user switches between "Flamegraph/Top Functions" tabs, the data is taken from the cache instead of making the same request.